cmd/nodecmd/create_devnet.go

```diff
@@ -188,40 +189,50 @@ func setupDevnet(clusterName string, hosts []*models.Host) error {
 	}
 	walletAddrStr := k.X()[0]
 
+	// exclude RPC nodes from genesis file generation as they will have no stake
+	hostsRPC := utils.Filter(hosts, func(h *models.Host) bool {
+		return slices.Contains(maps.Keys(rpcNodeIPMap), h.GetCloudID())
+	})
+	hostsWithoutRPC := utils.Filter(hosts, func(h *models.Host) bool {
+		return !slices.Contains(maps.Keys(rpcNodeIPMap), h.GetCloudID())
+	})
+	hostsWithoutRPCIDs := utils.Map(hostsWithoutRPC, func(h *models.Host) string { return h.NodeID })
+
 	// create genesis file at each node dir
-	genesisBytes, err := generateCustomGenesis(network.ID, walletAddrStr, stakingAddrStr, hosts)
+	genesisBytes, err := generateCustomGenesis(network.ID, walletAddrStr, stakingAddrStr, hostsWithoutRPC)
 	if err != nil {
 		return err
 	}
-	for _, cloudHostID := range cloudHostIDs {
-		outFile := filepath.Join(app.GetNodeInstanceDirPath(cloudHostID), "genesis.json")
-		if err := os.WriteFile(outFile, genesisBytes, constants.WriteReadReadPerms); err != nil {
-			return err
-		}
-	}
 
 	// create avalanchego conf node.json at each node dir
 	bootstrapIPs := []string{}
 	bootstrapIDs := []string{}
-	for i, ansibleHostID := range ansibleHostIDs {
-		cloudHostID := cloudHostIDs[i]
+	for _, host := range append(hostsWithoutRPC, hostsRPC...) {
```

Comment:
can you use hosts

Reply:
no, as hosts is unordered. In the current append() way, we make sure that hostsRPC get full list of validators to connect to, as we keep appending to node.json validator list as we go through the slice

Reply:
pls add a comment for this sorting req

Reply:
k